gateway-ha/src/main/java/io/trino/gateway/ha/router/QueryCountBasedRouter.java

```diff
@@ -224,7 +224,7 @@ private synchronized Optional<String> getBackendForRoutingGroup(String routingGr
     @Override
     public String provideAdhocBackend(String user)
     {
-        return getBackendForRoutingGroup("adhoc", user).orElseThrow();
+        return getBackendForRoutingGroup("adhoc", user).orElseThrow(() -> new ClusterNotFoundException("did not find any cluster for the adhoc routing group"));
```

Comment:
Should we make this more generic of an exception? Like RoutingException or so? Also should we pass in the magic `adhoc` from the prior code?

And update wording?

```
No cluster in routing group `adhoc` available.
```

Or similar

Reply:
we could put it the method `getBackendForRoutingGroup`. That method has the `routingGroup` passed in, so we can use that in our error message. That should be cleaner

Reply:
Open to making the error more generic to be `RoutingException`

Reply:
I think more general version, RoutingException is better :)